src/mgos_pcf857x.c

```diff
@@ -352,6 +372,45 @@ bool mgos_pcf857x_gpio_set_button_handler(struct mgos_pcf857x *dev, int pin, enu
   (void)pull_type;
 }
 
+void mgos_pcf857x_gpio_blink_cb(void *arg) {
+  struct mgos_pcf857x_gpio_blink_state *bs = (struct mgos_pcf857x_gpio_blink_state *)arg;
+
+  if (bs != NULL) {
+    bool curr = mgos_pcf857x_gpio_toggle(bs->dev, bs->pin);
+    if (bs->blink.on_ms != bs->blink.off_ms) {
+      int timeout = (curr ? bs->blink.on_ms : bs->blink.off_ms);
+      bs->blink.timer_id = mgos_set_timer(timeout, 0, mgos_pcf857x_gpio_blink_cb, bs);
+    }
+  }
+}
+
+bool mgos_pcf857x_gpio_blink(struct mgos_pcf857x *dev, int pin, int on_ms, int off_ms) {
+  bool res = !(!dev || on_ms < 0 || off_ms < 0 || on_ms >= 65536 || off_ms >= 65536);
+  if (res) {
+    struct mgos_pcf857x_gpio_blink_state *bs = mgos_pcf857x_get_or_create_blink_state(dev, pin);
```

Comment:
bs is never freed. your docs say if on_ms==off_ms==0, the timer will be cleaned up. Function below must clean up and free the bs struct then.